### PR TITLE
Jennyf/ws trust sovereign clouds

### PR DESCRIPTION
**Pull request by jennyf19**
#591 When logging in as a federated user, the default audience is urn:federation:MicrosoftOnline. However, this does not work for a sovereign cloud service. Code has been changed to use the JSON property cloud_audience_urn from the HTTPS GET response as the audience value. If no value is present, a default value will be used (urn:federation:MicrosoftOnline).

### CODE REVIEW

File: src/ADAL.PCL/AcquireTokenNonInteractiveHandler.cs [hunk 1]
```diff
@@ -111,11 +113,15 @@ public AcquireTokenNonInteractiveHandler(RequestData requestData, UserAssertion
                     {
                         throw new AdalException(AdalError.MissingFederationMetadataUrl);
                     }
+                    if (string.IsNullOrWhiteSpace(userRealmResponse.CloudAudienceUrn))
```
Comment (kpanwar) [inline src/ADAL.PCL/AcquireTokenNonInteractiveHandler.cs]: this if check and defaultAppliesTo variable should be inside the WsTrustRequest class only. Let the request class make the decision on how the value is used.

File: tests/Test.ADAL.NET.Unit/NonInteractiveTests.cs [hunk 2]
```diff
@@ -127,10 +127,60 @@ public void Initialize()
         }
 
         [TestMethod]
+        [Description("Cloud Audience Urn Test")]
+        public async Task CloudAudienceUrnTest()
```
Comment (kpanwar) [inline tests/Test.ADAL.NET.Unit/NonInteractiveTests.cs]: this should be 2 separate tests.

File: src/ADAL.PCL/AcquireTokenNonInteractiveHandler.cs (hunk 1 above)
Comment (jennyf19) [inline src/ADAL.PCL/AcquireTokenNonInteractiveHandler.cs]: Done

File: tests/Test.ADAL.NET.Unit/NonInteractiveTests.cs (hunk 2 above)
Comment (jennyf19) [inline tests/Test.ADAL.NET.Unit/NonInteractiveTests.cs]: Changes: "Cloud Audience Urn Test" to test the wsTrustRequest passing it a generic cloud audience urn (ex. "urn:federation:Blackforest"). "Cloud Audience Urn Null Test" to test wsTrustRequest passing it a null value for the cloud audience. 
Added a "Ws-Trust Request Test" to test WsTrustRequest, passing it a test constant value ("urn:federation:MicrosoftOnline"). Also, a "Ws-Trust Request Generic Cloud Urn Test, to test WsTrustRequest, passing it a test constant value ("urn:federation:Blackforest").

**Comment by kpanwar** [comment]
:shipit:

**Pull request merged**